File: C1_portail/objects/carnet_maintenance/E_Model_to_Nom.php
<?php
/**
 * Class d'objet EModele
 * - Contient tous les modèles de bateau
 */

require_once('MyEnum.php');

abstract class E_Model_to_Nom extends MyEnum
{
    const __default = self::default;
    const default = "default";
    const voilier = "voilier";
    const trois_mat = "trois_mat";
    const moteur = "moteur";
    const sportif = "sportif";
    const zodiac = "zodiac";
    const croisiere = "croisiere";
}

File: B_back_office/objects/E_Model_to_Nom.java
package B_back_office.objects;

import org.json.JSONArray;

public enum E_Model_to_Nom {

    VOILIER("voilier"),
    TROISMAT("troismat"),
    MOTEUR("moteur"),
    SPORTIF("sportif"),
    ZODIAC("zodiac"),
    CROISIERE("croisiere");

    private final String value;

    E_Model_to_Nom(final String value) {
        this.value = value;
    }

    public static JSONArray toJsonArray() {
        JSONArray ja = new JSONArray();
        for (E_Model_to_Nom pair : E_Model_to_Nom.values()) {
            ja.put(pair.value);
        }
        return ja;
    }

    public static void main(String[] args) {
        System.out.println(toJsonArray());
    }
}